x/datapool/keeper/pool.go

```diff
@@ -163,3 +187,92 @@ func (k Keeper) SetPool(ctx sdk.Context, pool *types.Pool) {
 	bz := k.cdc.MustMarshalBinaryLengthPrefixed(pool)
 	store.Set(poolKey, bz)
 }
+
+func (k Keeper) SetContractAddress(ctx sdk.Context, address sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyContractAddress, address)
+}
+
+func (k Keeper) GetContractAddress(ctx sdk.Context) (sdk.AccAddress, error) {
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(types.KeyContractAddress) {
+		return nil, sdkerrors.Wrapf(types.ErrNoRegisteredContract, "no contract registered")
+	}
+
+	return store.Get(types.KeyContractAddress), nil
+}
+
+func (k Keeper) CreateContract(ctx sdk.Context, creator sdk.AccAddress, wasmCode []byte) (uint64, error) {
+	// access configuration of only for creator address
+	accessConfig := &wasmtypes.AccessConfig{
+		Permission: wasmtypes.AccessTypeOnlyAddress,
+		Address:    creator.String(),
+	}
+
+	// create contract
+	codeID, err := k.wasmKeeper.Create(ctx, creator, wasmCode, accessConfig)
+	if err != nil {
+		return 0, sdkerrors.Wrapf(err, "failed to create contract")
+	}
+
+	return codeID, nil
+}
+
+// DeployAndRegisterContract creates, instantiate contract and store contract address
+func (k Keeper) DeployAndRegisterContract(ctx sdk.Context, wasmCode []byte) error {
+	moduleAddr := types.GetModuleAddress()
```

Comment:
Thank you. I didn't know this module account. It would be better to declare module address in `app.go` as you suggested. nice!

Reply:
The module account address is generated by the same way of other module account address. ([e0fe83f](https://github.com/medibloc/panacea-core/pull/288/commits/e0fe83f6c1350b66852643878efdd2e1237d766a))
https://github.com/medibloc/panacea-core/blob/7595cf300067e0e2969643cbb5a15acbf7960912/app/app.go#L633

Also, add to `maccPerms` as you mentioned to prevent from sending funds to module account. I knew that the `maccPerms` is used for not only declaration of module accounts' permission but prevention of sending funds to module account.

Reply:
👍

Reply:
👍